For splitting binutils-<nativetriplet> out from binutils, we need to ensure
that no symlink from a triplet prefixed name goes to a non-triplet prefixed
name. Due to the renaming technique in use, we'd previously ended up with:

 * <triplet>-ld -> ld.bfd, but it should go to <triplet>-ld.bfd
 * gold -> ld.gold, but it should go to <triplet>-gold
 * missing <triplet>-gold, but it should go to <triplet>-ld.gold

Index: debian/rules
===================================================================
--- debian.orig/rules
+++ debian/rules
@@ -308,7 +308,7 @@
 programs = addr2line ar as c++filt elfedit gprof ld ld.bfd
 programs += nm objcopy objdump ranlib readelf size strings strip
 ifeq ($(with_gold),yes)
-programs += ld.gold dwp
+programs += gold ld.gold dwp
 endif
 
 ########################################
@@ -1120,12 +1120,10 @@
 	mv $(d_bin)/$(PF)/bin/ld $(d_bin)/$(PF)/bin/ld.bfd
 	mv $(d_bin)/$(PF)/share/man/man1/ld.1 \
 		$(d_bin)/$(PF)/share/man/man1/ld.bfd.1
-	ln -sf ld.bfd $(d_bin)/$(PF)/bin/ld
-	ln -sf $(DEB_HOST_GNU_TYPE)-ld.bfd $(d_bin)/$(PF)/bin/$(DEB_HOST_GNU_TYPE)-ld
-	ln -sf ld.bfd.1.fz $(d_bin)/$(PF)/share/man/man1/ld.1.gz
-	ln -sf $(DEB_HOST_GNU_TYPE)-ld.bfd.1.gz $(d_bin)/$(PF)/share/man/man1/$(DEB_HOST_GNU_TYPE)-ld.1.gz
+	ln -sf $(DEB_HOST_GNU_TYPE)-ld.bfd $(d_bin)/$(PF)/bin/ld
+	ln -sf $(DEB_HOST_GNU_TYPE)-ld.bfd.1.fz $(d_bin)/$(PF)/share/man/man1/ld.1.gz
 ifeq ($(with_gold),yes)
-	ln -s ld.gold $(d_bin)/$(PF)/bin/gold
+	ln -s $(DEB_HOST_GNU_TYPE)-ld.gold $(d_bin)/$(PF)/bin/gold
   ifeq ($(DEB_BUILD_GNU_TYPE),$(DEB_HOST_GNU_TYPE))
 	if which help2man >/dev/null 2>&1; then \
 	  help2man -N -n 'The GNU ELF linker' $(d_bin)/$(PF)/bin/ld.gold \
@@ -1136,7 +1134,7 @@
   endif
 	cp debian/dwp.1 $(d_bin)/$(PF)/share/man/man1/
 	cp debian/ld.gold.1 $(d_bin)/$(PF)/share/man/man1/
-	ln -s ld.gold.1.gz $(d_bin)/$(PF)/share/man/man1/gold.1.gz
+	ln -s $(DEB_HOST_GNU_TYPE)-ld.gold.1.gz $(d_bin)/$(PF)/share/man/man1/gold.1.gz
 
 	: # install a symlink for the gold linker
 	$(install_dir) $(d_bin)/$(PF)/lib/gold-ld
@@ -1154,7 +1152,11 @@
 	for i in $(programs); do \
 	  mv $(d_bin)/$(PF)/bin/$$i $(d_bin)/$(PF)/bin/$(DEB_HOST_GNU_TYPE)-$$i; \
 	  ln -sf $(DEB_HOST_GNU_TYPE)-$$i $(d_bin)/$(PF)/bin/$$i; \
-	  mv $(d_bin)/$(PF)/share/man/man1/$$i.1 $(d_bin)/$(PF)/share/man/man1/$(DEB_HOST_GNU_TYPE)-$$i.1; \
+	  if test -L $(d_bin)/$(PF)/share/man/man1/$$i.1.gz; then \
+	    mv $(d_bin)/$(PF)/share/man/man1/$$i.1.gz $(d_bin)/$(PF)/share/man/man1/$(DEB_HOST_GNU_TYPE)-$$i.1.gz; \
+	  else \
+	    mv $(d_bin)/$(PF)/share/man/man1/$$i.1 $(d_bin)/$(PF)/share/man/man1/$(DEB_HOST_GNU_TYPE)-$$i.1; \
+	  fi; \
 	  ln -sf $(DEB_HOST_GNU_TYPE)-$$i.1.gz $(d_bin)/$(PF)/share/man/man1/$$i.1.gz; \
 	done
 ifeq ($(with_multiarch),yes)
